crates/starknet/src/starknet/mod.rs

```diff
@@ -231,6 +336,10 @@ impl Starknet {
         // clear pending block information
         self.generate_pending_block()?;
 
+        if self.config.dump_on.is_some() && self.config.dump_on == Some(DumpMode::OnTransaction) {
+            self.dump_transactions();
```

Comment:
Won't we have duplicate transactions with this? Can't see any logic that tracks what transactions were already dumped

Reply:
The current implementation is just dumping them all into a file on every new transaction.

After a daily talk today, we decided to handle this optimization in a new issue https://github.com/0xSpaceShard/starknet-devnet-rs/issues/175 

Also is still not clear if re-execution will be the final implementation of dump and load.

Reply:
The new update is that even with blockier we will do re-execution optimization can be done in this PR.